File: Python/0011-m-container-with-most-water.py
"""
Time: O(n)
Space: O(1) 
"""
class Solution:
    def maxArea(self, height: List[int]) -> int:
        size = len(height)
        maxm = 0
        j = 0
        k = size - 1
        while j < k:
            if height[j] <= height[k]:
                maxm = max(maxm, height[j] * (k - j))
                j += 1
            else:
                maxm = max(maxm, height[k] * (k - j))
                k -= 1
        return maxm

File: leetcode/C++/0011-m-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0, j = height.size() - 1;
        int cur = 0, res = 0;
        while (i < j) {
            cur = (j - i) * min(height[i], height[j]);
            res = max(cur, res);
            if (height[i] < height[j])
                i++;
            else
                j--;
        }
        return res;
    }
};